### DISCUSSION TITLE
[@types/google.accounts] has incorrect type for `renderButton` width

### DISCUSSION THREAD

**Discussion by harperiscoding** [DefinitelyTyped/DefinitelyTyped#66192] [Issues with a @types package]
Hello!

When calling `google.accounts.id.renderButton` and passing in the an argument for `options` with `width`, this package expects `width` to be a `string`, but `renderButton` will throw an error if you pass it a string for width. The documentation is incorrect here: https://developers.google.com/identity/gsi/web/reference/js-reference#width

As you can see, google says this should be a `string` but provides a `number` as an example. Similarly, if you look into the code itself (it's minified so it's hard to read, but `c` is `width` here), the script throws an error if it receives a non-number value for `width`. 
```
        _.$e = function(a, b, c) {
            if (null != c && "number" !== typeof c)
                throw Error();
            return _.M(a, b, c)
        }
```

It's possible this was a recent change to the google script, but it did break our GSI flows yesterday— would be great to get an update here!

[a comment by an automated account is not shown]

**Comment by mateuszJS** [comment]
Same experience as @walkerspence , my research confirms the behaviour described above. Moreover it throws an error without any message... Not the easiest way to figure out what is going on 😅  The change is needed in this place: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/google.accounts/index.d.ts#L629C1-L633C28

**Reply by harperiscoding** [reply]
Were you able to confirm that the script changed on Google's end?

**Comment by PartyLich** [comment]
Thank you for highlighting this. A PR was opened by kilinkis and has been approved. 

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66207
